google/cloud/pubsublite/internal/partition_publisher.cc

```diff
@@ -159,11 +168,30 @@ void PartitionPublisher::OnRead(absl::optional<PublishResponse> response) {
                "Server sent message response when no batches were "
                "outstanding."));
   }
-  std::int64_t offset = response->message_response().start_cursor().offset();
-  for (auto& message : batch) {
+
+  // ensure cursor ranges are sorted by increasing message batch index
+  auto* ranges = response->mutable_message_response()->mutable_cursor_ranges();
+  std::sort(ranges->begin(), ranges->end(),
+            [](CursorRange const& a, CursorRange const& b) {
+              return a.start_index() < b.start_index();
+            });
+  int range_idx = 0;
```

Comment:
It would help if you could explain what these loops are trying to do.  I am sure it is obvious to somebody familiar with Pub/Sub Lite, but it is hard to glean what is the intent, and therefore hard to offer useful comments.